DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/mmhost/src/mmhost.cpp

```diff
@@ -0,0 +1,149 @@
+  auto selector = sycl::ext::intel::fpga_selector_v;
+#else  // #if FPGA_EMULATOR
+  auto selector = sycl::ext::intel::fpga_emulator_selector_v;
+#endif
+
+  try {
+    sycl::queue q(selector, fpga_tools::exception_handler,
+                  sycl::property::queue::enable_profiling{});
+
+    // Print out the device information.
+    std::cout << "Running on device: "
+              << q.get_device().get_info<sycl::info::device::name>() << "\n";
+
+    // Create and initialize the host arrays
+    constexpr int kN = 8;
+    std::cout << "elements in vector : " << kN << "\n";
+
+    auto host_array_A =
+        malloc_shared<int>(kN, q, property_list{usm_buffer_location(kBL1)});
+    assert(host_array_A);
+    auto host_array_B =
+        malloc_shared<int>(kN, q, property_list{usm_buffer_location(kBL2)});
+    assert(host_array_B);
+    auto host_array_C =
+        malloc_shared<int>(kN, q, property_list{usm_buffer_location(kBL3)});
+    assert(host_array_C);
+
+    auto pointer_array_A = malloc_shared<int>(kN, q);
+    assert(pointer_array_A);
+    auto pointer_array_B = malloc_shared<int>(kN, q);
+    assert(pointer_array_B);
+    auto pointer_array_C = malloc_shared<int>(kN, q);
+    assert(pointer_array_C);
+
+    for (int i = 0; i < kN; i++) {
+      host_array_A[i] = i;
+      host_array_B[i] = i * 2;
+
+      pointer_array_A[i] = i;
+      pointer_array_B[i] = i * 2;
+    }
+
+    // Run the kernal code
+    q.single_task(MMHostIP{host_array_A, host_array_B, host_array_C, kN})
+        .wait();
+    q.single_task(
+         PointerIP{pointer_array_A, pointer_array_B, pointer_array_C, kN})
+        .wait();
+
+    // Check to see if results are correct
+    bool passed = true;
+    for (int i = 0; i < kN; i++) {
+      auto golden = host_array_A[i] + host_array_B[i];
+      if (host_array_C[i] != golden || pointer_array_C[i] != golden) {
+        std::cout << "ERROR! At index: " << i << " , expected: " << golden
+                  << " , found: " << pointer_array_C[i] << "\n";
+        passed = false;
+      }
+    }
+
+    if (passed) {
+      std::cout << "--> PASS"
+                << "\n";
+    }
+
+    // Free memory
+    free(host_array_A, q);
+    free(host_array_B, q);
+    free(host_array_C, q);
+    free(pointer_array_A, q);
+    free(pointer_array_B, q);
+    free(pointer_array_C, q);
+
+  } catch (sycl::exception const &e) {
+    // Catches exceptions in the host code
+    std::cerr << "Caught a SYCL host exception:\n" << e.what() << "\n";
+
+    // Most likely the runtime couldn't find FPGA hardware!
+    if (e.code().value() == CL_DEVICE_NOT_FOUND) {
+      std::cerr << "If you are targeting an FPGA, please ensure that your "
+                   "system has a correctly configured FPGA board.\n";
+      std::cerr << "Run sys_check in the oneAPI root directory to verify.\n";
+      std::cerr << "If you are targeting the FPGA emulator, compile with "
+                   "-DFPGA_EMULATOR.\n";
+    }
+    std::terminate();
```

Comment:
I thought we are not required to do this anymore, confirm with Yohann.

Reply:
Alternatively you can confirm by checking other tutorials if they still have this exception section.